### ISSUE TITLE
resume=true parameter creating new directory instead of appending to existing directory

### ISSUE THREAD

**Issue by Kylamber** [ultralytics/ultralytics#21963]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi! I have noticed that v8.3.193's resume directory saving behavior changed from v8.3.192's. Usually it would append the resume result in the model's directory.

Here is the code to reproduce it. First I start a training process and wait for 1 epoch.

```python
from ultralytics import YOLO

model = YOLO("yolov11n.pt")

results = model.train(data="coco.yaml", epochs=1200, imgsz=640, name="train")
```

This would create a directory `runs/detect/train`. Then, resume the training

```python
from ultralytics import YOLO

model = YOLO("runs/detect/train/weights/last.pt")

results = model.train(resume=True)
```

This would create a new directory `runs/detect/train2` instead of appending the results to `runs/detect/train`. Is this the intended behavior? I have tried using the parameter `exist_ok=True` and it still creates a new directory.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Kylamber, thank you for your interest in Ultralytics 🚀! This is an automated response to help move things forward quickly. An Ultralytics engineer will also assist you here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, thanks for the clear steps you shared. To help us reproduce precisely, please also include:
- Full console logs from start to resume (including any warnings)  
- Your OS, Python, PyTorch, CUDA versions, and GPU model  
- Exact `ultralytics` package version (e.g., 8.3.193) and how it was installed  
- The full command or script used for both the initial run and resume  
If you have a smaller script that isolates the behavior, a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) is appreciated.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue is now merged in PR [Restore all arguments on checkpoint load (#21968)](https://github.com/ultralytics/ultralytics/pull/21968). This change preserves the full set of checkpoint arguments on load, removing the previous filtering and ensuring `model.args` includes all metadata. This should improve compatibility with custom configs, training resumption, exports, and Ultralytics HUB workflows. Note: downstream code that inspects `model.args` may now see additional keys and should handle them gracefully.

Credit to the author Y-T-G with contributions from Laughing-q — thank you for the improvement!

Ways to test the fix:
- Install the latest from main:
  ```
  pip install -U git+https://github.com/ultralytics/ultralytics.git@main
  ```
- Or wait for the next official PyPI release and upgrade then:
  ```
  pip install -U ultralytics
  ```
- Quick verification snippet:
  ```python
  from ultralytics import YOLO

  model = YOLO("path/to/your/checkpoint.pt")
  print(model.args)  # should include all original/custom args from the checkpoint
  ```

Feedback on whether this resolves the issue would be appreciated. Thanks for reporting this, and please share any remaining problems or edge cases encountered. 🙏
